vizro-core/src/vizro/models/types.py

```diff
@@ -203,19 +206,43 @@ def _parse_json(
             raise ValueError(f"_target_={function_name} cannot be imported from {import_path.__name__}.") from exc
 
         # All the other items in figure are the keyword arguments to pass into function.
-        function_kwargs = callable_config
+        function_kwargs = captured_callable_config
 
         # It would seem natural to return cls(function, **function_kwargs) here, but the function is already decorated
         # with @capture, and so that would return a nested CapturedCallable.
-        captured_callable = function(**function_kwargs)
```

Comment:
Having this logic duplicated between here and lines 177 onwards was always very unsatisfactory. Now I've detangled the validation into multiple validators it can be removed.